homeassistant/components/script.py

```diff
@@ -42,6 +44,62 @@
 
 _LOGGER = logging.getLogger(__name__)
 
+_ALIAS_VALIDATOR = vol.Schema(cv.string)
+
+
+def _alias_stripper(validator):
+    """Strip alias from object for validation."""
+    def validate(value):
+        """Validate without alias value."""
+        value = value.copy()
+        alias = value.pop(CONF_ALIAS, None)
+
+        if alias is not None:
+            alias = _ALIAS_VALIDATOR(alias)
+
+        value = validator(value)
+
+        if alias is not None:
+            value[CONF_ALIAS] = alias
+
+        return value
+
+    return validate
+
+
+_DELAY_SCHEMA = {
+    vol.Required(CONF_DELAY): vol.All({
+        CONF_ALIAS: str,
+        'days': vol.All(vol.Coerce(int), vol.Range(min=0)),
+        'seconds': vol.All(vol.Coerce(int), vol.Range(min=0)),
+        'microseconds': vol.All(vol.Coerce(int), vol.Range(min=0)),
+        'milliseconds': vol.All(vol.Coerce(int), vol.Range(min=0)),
+        'minutes': vol.All(vol.Coerce(int), vol.Range(min=0)),
+        'hours': vol.All(vol.Coerce(int), vol.Range(min=0)),
+        'weeks': vol.All(vol.Coerce(int), vol.Range(min=0)),
+    }, cv.has_at_least_one_key([
+        'days', 'seconds', 'microseconds', 'milliseconds', 'minutes', 'hours',
+        'weeks']))
+}
+
+_EVENT_SCHEMA = cv.EVENT_SCHEMA.extend({
+    CONF_ALIAS: cv.string,
+})
+
+_SCRIPT_ENTRY_SCHEMA = vol.Schema({
+    CONF_ALIAS: str,
+    vol.Required(CONF_SEQUENCE): vol.All(vol.Length(min=1), [vol.Any(
```

Comment:
That point is exactly what I was struggling with for a couple of hours last night. When you specify a list you say: I expect a list and this are the valid values. However, if I omit `vol.Any` I get validation errors when the first schema doesn't pass (because the second type is valid).

However, I am unable to repro it with simple examples on the command line right now :( If you want you can have a stab at it later to simplify it. Tests will tell you when things don't work.